libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -32,101 +30,120 @@ impl<'a> LiftAstToDml<'a> {
     /// the attributes defined by the given sources registered.
     ///
     /// The attributes defined by the given sources will be namespaced.
-    pub(crate) fn new(
-        source: Option<&'a configuration::Datasource>,
-        preview_features: &HashSet<PreviewFeature>,
-        db: &'a ParserDatabase<'a>,
-    ) -> LiftAstToDml<'a> {
+    pub(crate) fn new(preview_features: &HashSet<PreviewFeature>, db: &'a ParserDatabase<'a>) -> LiftAstToDml<'a> {
         LiftAstToDml {
             attributes: AllAttributes::new(preview_features),
-            source,
             db,
         }
     }
 
-    pub fn lift(&self) -> Result<dml::Datamodel, Diagnostics> {
+    pub fn lift(&self, diagnostics: &mut Diagnostics) -> dml::Datamodel {
```

Comment:
What about having the `&mut Diagnostics` in the constructor, then calling this with `self` by-value? Would cut one parameter out from the methods/functions...

Reply:
That's one path we can go. The other would be having `lift()` as a free function. Wdyt? I'm not sure.

Reply:
Some of the functions have way too many params. Clippy says 7 or 8 is the maximum, but 5 already looks like a lot to my eyes.

Reply:
We'll go with that :+1: Commit incoming.